### PR TITLE
Documentation for `TreeDecorator` & `TreeDecoratorService`

### LINKED ISSUE #8696: Add/refine documentation for `TreeDecorator`
Add API documentation for `TreeDecorator` and potentially an extension guide. This will help developers to better understand how to provide custom decorators.

### PR DESCRIPTION
**Pull request by tortmayr**
Fixes #8696
- Refine documentation for `TreeDecorator`
- Refine documentation for `TreeDecoratorService`
Signed-off-by: Tobias Ortmayr <tortmayr@eclipsesource.com>
Contributed on behalf of STMicroelectronics

#### What it does

Add/refines documentation for TreeDecorator.
#### How to test

Documentation changes only
#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/browser/tree/tree-decorator.ts [hunk 1]
```diff
@@ -30,25 +62,41 @@ export interface TreeDecorator {
...
+export const TreeDecoratorService = Symbol('TreeDecoratorService');
+
 /**
- * Decorator service which emits events from all known tree decorators.
- * Keys are the unique tree node IDs and the values
- * are the decoration data collected from all the decorators known by this service.
+ * The {@link TreeDecoratorService} manages a set of known {link TreeDecorator}s and emits events when
+ * any of the known decorators has changes. Typically, a `TreeDecoratorService` provides a contribution point that can be used to
+ * register {@link TreeDecorator}s exclusively for this service.
+ * e.g. the {@link NavigatorDecoratorService} provides the  {@link NavigatorTreeDecorator} contribution point to register custom decorators.
```
Comment (vince-fugnitto) [inline packages/core/src/browser/tree/tree-decorator.ts:86]: I think we can exclude this comment, `NavigatorDecoratorService` does not exist in `@theia/core`.

**Review by vince-fugnitto — COMMENTED** [review]
The overall content looks good :+1:

File: packages/core/src/browser/tree/tree-decorator.ts (hunk 1 above)
Comment (tortmayr) [reply packages/core/src/browser/tree/tree-decorator.ts]: Done

**Comment by tortmayr** [comment]
Thanks for the fast review. I applied your feedback suggestions.

**Review by vince-fugnitto — APPROVED** [review]
The changes look good :+1: 
Please be sure to squash and include a proper `sign-off` (currently failing the eca check) so we can merge.

**Pull request merged**